vignettes/understanding_shapr_vaeac.Rmd

```diff
@@ -0,0 +1,981 @@
+  model = model,
+  x_explain = x_explain,
+  x_train = x_train,
+  approach = approach,
+  prediction_zero = p0,
+  n_combinations = n_combinations,
+  group = group,
+  n_samples = n_samples,
+  n_batches = n_batches,
+  seed = seed,
+  vaeac.num_vaeacs_initiate = vaeac.num_vaeacs_initiate,
+  vaeac.epochs = 7, # Duplicate
+  vaeac.verbose = FALSE, # Should have been an extra parameter
+  vaeac.extra_parameters = list(
+    vaeac.epochs = 10, # Duplicate, will use this one.
+    vaeac.depth = 2, # Should have been a separate parameter, will use this one
+    vaeac.save_every_nth_epoch = vaeac.save_every_nth_epoch, # Correct place
+    what_is_this_parameter = 4 # Unknown parameter
+  )
+)
+
+# Look at the used parameters that was misplaced
+expl_misplaced_parameters$internal$parameters[c("vaeac.epochs", "vaeac.verbose", "vaeac.depth")]
+```
+
+
+## Paired sampling
+
+The `vaeac` approach use what is called paired sampling to improve the stability of the training procedure.
+That is, each training batch contains two versions of the same training observation, but where the first one is
+masked by \eqn{S} and the second one is masked by \eqn{\bar{S}}, the complement, see
+\href{https://arxiv.org/pdf/2107.07436.pdf}{Jethani et al. (2022)}. 
+The training time will in due to more complex implementation in comparison to random sampling.
+
+```{r paired sampling training, cache = TRUE}
+expl_paired_sampling_TRUE <- explain(
+  model = model,
+  x_explain = x_explain,
+  x_train = x_train,
+  approach = approach,
+  prediction_zero = p0,
+  n_combinations = n_combinations,
+  n_samples = n_samples,
+  n_batches = n_batches,
+  seed = seed,
+  vaeac.epochs = 100,
+  vaeac.num_vaeacs_initiate = 5,
+  vaeac.extra_parameters = list(
+    vaeac.verbose = FALSE,
+    vaeac.paired_sampling = TRUE
+  )
+)
+
+expl_paired_sampling_FALSE <- explain(
+  model = model,
+  x_explain = x_explain,
+  x_train = x_train,
+  approach = approach,
+  prediction_zero = p0,
+  n_combinations = n_combinations,
+  n_samples = n_samples,
+  n_batches = n_batches,
+  seed = seed,
+  vaeac.epochs = 100,
+  vaeac.num_vaeacs_initiate = 5,
+  vaeac.extra_parameters = list(
+    vaeac.verbose = FALSE,
+    vaeac.paired_sampling = FALSE
+  )
+)
+```
+
+We can compare the results by looking at the training and validation errors and by the $\opertorname{MSE}_v$ evaluation criterion. 
+```{r paired sampling plotting, cache = TRUE}
+explanation_list <- list("Regular samp." = expl_paired_sampling_FALSE, "Paired samp." = expl_paired_sampling_TRUE)
+plot_several_vaeacs_VLB_IWAE(explanation_list)
+plot_MSEv_eval_crit(explanation_list)
+```
+
+By looking at the time, we see that the paired version takes longer time in the setup_computation
+phase, that is, in the training phase.
+```{r paired sampling timing}
+rbind(
+  "time_paired_sampling" = expl_paired_sampling_TRUE$timing$timing_secs,
+  "time_regular_sampling" = expl_paired_sampling_FALSE$timing$timing_secs
+)
+```
+
+
+
+## Progressr
```

Comment:
I have kept it since we talked about verbose as a parameter to explain. I think this is something we can come back to in a another PR where we define what verbose = 1 and verbose = 2 actually means. Now I have set everything to verbose == 2, and I use message. Could use progressr with sticky, but then we need to be sure that a progress bar exists. So we should rather come back to this.